src/eh1/delay.rs

```diff
@@ -150,24 +145,12 @@ impl Transaction {
 /// MockDelay transaction kind.
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub enum TransactionKind {
-    /// Delay in nanoseconds
-    DelayNs(u32),
-    /// Asynchronous delay in nanoseconds
-    AsyncDelayNs(u32),
     /// Any delay in nanoseconds, blocking or async
-    AnyDelayNs(u32),
-    /// Delay in microseconds
-    DelayUs(u32),
-    /// Asynchronous delay in microseconds
-    AsyncDelayUs(u32),
-    /// Any delay in microseconds, blocking or async
-    AnyDelayUs(u32),
-    /// Delay in milliseconds
-    DelayMs(u32),
-    /// Asynchronous delay in milliseconds
-    AsyncDelayMs(u32),
-    /// Any delay in milliseconds, blocking or async
-    AnyDelayMs(u32),
+    Delay(u64),
+    /// Delay in nanoseconds, must be blocking
+    BlockingDelay(u64),
+    /// Delay in nanoseconds, must be async
+    AsyncDelay(u64),
```

Comment:
Nit: While I really like the simplification, I think that for public APIs the time unit shoulud be obvious, and not just be part of the docs.

Two ideas:

- `DelayNs` / `BlockingDelayNs` / `AsyncDelayNs` (my preferred variant)
- `Delay { ns: u64 }` / `BlockingDelayNs { ns: u64 }` / `AsyncDelayNs { ns: u64 }`

Reply:
While `TransactionKind` is not really intended to be used directly by the end user, I agree that the proposed naming is more explicit.